packages/sdk/src/client.ts

```diff
@@ -607,34 +728,118 @@ export class OmnisolClient {
 
   async liquidateCollateral(props: LiquidateCollateralProps) {
     const payer = this.wallet.publicKey
-    const [userOwner] = await this.pda.user(props.userOwner)
     const userWallet = props.userWallet
+    const collateralOwnerWallet = props.collateralOwnerWallet
+    const pool = props.pool
+
+    const [collateralOwner] = await this.pda.user(collateralOwnerWallet)
     const [user] = await this.pda.user(userWallet)
-    const [collateral] = await this.pda.collateral(props.sourceStake, userOwner)
-    const [poolAuthority] = await this.pda.poolAuthority(props.pool)
+    const [collateral] = await this.pda.collateral(props.sourceStake, collateralOwner)
+    const [poolAuthority] = await this.pda.poolAuthority(pool)
+    const [liquidator] = await this.pda.liquidator(payer)
+    const [oracle] = await this.pda.oracle()
+
     const userData = await this.fetchUser(user)
-    const withdrawIndex = userData.lastWithdrawIndex - userData.requestsAmount
-    const [withdrawInfo] = await this.pda.withdrawInfo(payer, withdrawIndex + 1)
+    const withdrawIndex = userData.lastWithdrawIndex - userData.requestsAmount + 1
+    const [withdrawInfo] = await this.pda.withdrawInfo(payer, withdrawIndex)
+
+    const collateralData = await this.fetchCollateral(collateral)
+    let anchorRemainingAccounts: web3.AccountMeta[]
+    if (collateralData.isNative) {
+      const splitStake = props.splitStake ?? web3.Keypair.generate().publicKey
+      anchorRemainingAccounts = [{
+        pubkey: splitStake,
+        isWritable: true,
+        isSigner: true,
+      }]
+    } else {
+      const stakePool = props.stakePool ?? web3.Keypair.generate().publicKey
```

Comment:
No need to use `web3.Keypair.generate().publicKey` here... `stakePool` should be required prop. Please check other props

Reply:
Pubkey.default()